src/GraphQL/Extensions/SchemaExtensions.cs

```diff
@@ -81,12 +84,23 @@ public static void RegisterTypes<TSchema>(this TSchema schema, params IGraphType
         /// </summary>
         /// <typeparam name="TClrType">The CLR property type from which to infer the GraphType.</typeparam>
         /// <typeparam name="TGraphType">Inferred GraphType.</typeparam>
-        public static void RegisterTypeMapping<TClrType, TGraphType>(this ISchema schema)
+        public static void RegisterTypeMapping<TClrType,
+#if NET5_0_OR_GREATER
+            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)]
+#endif
+        TGraphType>(this ISchema schema)
             where TGraphType : IGraphType
         {
+            Preserve<GraphQLClrInputTypeReference<TClrType>>();
+            Preserve<GraphQLClrOutputTypeReference<TClrType>>();
             schema.RegisterTypeMapping(typeof(TClrType), typeof(TGraphType));
         }
 
+        /// <summary>
+        /// Prevents <typeparamref name="T"/> from being trimmed by the linker.
```

Comment:
How it works? After all, the method does not use any API from T so trimmer can remove everything from T while keeping the T class itself. I don't understand.

Reply:
Correct, but we don't need any constructors on `GraphQLClrInputReference<T>` since it is never constructed; in fact the constructor is private.  It's just a reference; we only need to be able to create the `Type` itself via `MakeGenericType`.

Reply:
WOW! It is just a type container. I completely forgot about it. OK, for `GraphQLClrInputReference/GraphQLClrOutputTypeReference` I got it but what about all other generic arguments passed into `Preserve`?

Reply:
Please reference the exact code so I can comment.  I believe all remaining instances of `Preserve<T>` only relate to the CLR type reference types, and I have removed the rest.

Reply:
OK, if something else remains I'll write.